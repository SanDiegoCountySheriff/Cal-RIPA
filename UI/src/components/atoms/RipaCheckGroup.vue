<template>
  <div class="ripa-check-group">
    <v-checkbox
      v-for="(item, index) in items"
      :key="index"
      v-model="model"
      :disabled="item.disabled || false"
      :label="item.name"
      :value="item.value"
      hide-details
    ></v-checkbox>
  </div>
</template>

<script>
export default {
  name: 'ripa-check-group',

  data() {
    return {
      viewModel: this.value,
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newVal) {
        this.viewModel = newVal
        this.$emit('input', newVal)
      },
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = newVal
    },
  },

  props: {
    value: {
      type: Array,
      default: () => [],
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
}
</script>
