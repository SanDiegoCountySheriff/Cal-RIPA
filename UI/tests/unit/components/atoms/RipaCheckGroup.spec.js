import RipaCheckGroup from '@/components/atoms/RipaCheckGroup.vue'
import { shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Check Group', () => {
  let vuetify
  let wrapper = null

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaCheckGroup, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  const testItems = [
    { name: 'Item 1', value: 'Item 1' },
    { name: 'Item 2', value: 'Item 2' },
  ]

  it('should match snapshot', () => {
    wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit event whem model changes', async () => {
    wrapper = factory({ items: testItems })

    wrapper.vm.model = testItems[0].value
    await wrapper.vm.$nextTick()

    expect(wrapper.emitted('input')).toBeTruthy()
    expect(wrapper.emitted('input').length).toBe(1)
  })

  it('should watch value', async () => {
    wrapper = factory()

    wrapper.setProps({ value: ['Test1', 'Test2'] })
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual(['Test1', 'Test2'])
  })
})
