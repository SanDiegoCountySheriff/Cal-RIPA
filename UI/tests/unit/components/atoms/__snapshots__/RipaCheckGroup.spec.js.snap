// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Check Group should match snapshot 1`] = `
<div
  class="ripa-check-group"
/>
`;
